nucypher/policy/conditions/json/rpc.py

```diff
@@ -0,0 +1,177 @@
+from abc import ABC
+from typing import Any, Optional, Tuple
+
+from marshmallow import ValidationError, fields, post_load, validate, validates
+from marshmallow.fields import Url
+from typing_extensions import override
+
+from nucypher.policy.conditions.context import is_context_variable
+from nucypher.policy.conditions.exceptions import (
+    JsonRequestException,
+)
+from nucypher.policy.conditions.json.base import (
+    HTTPMethod,
+    JSONPathField,
+    JsonRequestCall,
+)
+from nucypher.policy.conditions.json.utils import process_result_for_condition_eval
+from nucypher.policy.conditions.lingo import (
+    ConditionType,
+    ExecutionCallAccessControlCondition,
+    ReturnValueTest,
+)
+
+
+class BaseJsonRPCCall(JsonRequestCall, ABC):
+    class Schema(JsonRequestCall.Schema):
+        method = fields.Str(required=True)
+        params = fields.Field(required=False, allow_none=True)
+        query = JSONPathField(required=False, allow_none=True)
+        authorization_token = fields.Str(required=False, allow_none=True)
+
+        @validates("authorization_token")
+        def validate_auth_token(self, value):
+            if value and not is_context_variable(value):
+                raise ValidationError(
+                    f"Invalid value for authorization token; expected a context variable, but got '{value}'"
+                )
+
+    def __init__(
+        self,
+        method: str,
+        params: Optional[Any] = None,
+        query: Optional[str] = None,
+        authorization_token: Optional[str] = None,
+    ):
+        self.method = method
+        self.params = params or []
+
+        parameters = {
+            "jsonrpc": "2.0",
+            "method": self.method,
+            "params": self.params,
+            "id": 1,  # any id will do
+        }
+        super().__init__(
+            http_method=HTTPMethod.POST,
+            parameters=parameters,
+            query=query,
+            authorization_token=authorization_token,
+        )
+
+    @override
+    def _execute(self, endpoint, **context):
+        data = self._fetch(endpoint, **context)
+
+        # response contains a value for either "result" or "error"
+        error = data.get("error", None)
+        if error:
+            raise JsonRequestException(
+                f"JSON RPC Request failed with error in response: code={error['code']}, msg={error['message']}"
+            )
+
+        # obtain result first then perform query
+        result = data["result"]
```

Comment:
Should we catch here the case for a malformed JSON-RPC response that doesn't have a `result` field? Otherwise, we're left with a mere `KeyError`.

Reply:
Good catch - ✔️